src/shrinkage.jl

```diff
@@ -0,0 +1,84 @@
+"""
+    CoefByGroup
+
+Coefficients for groups of observations defined by levels of a grouping factor
+
+Fields include:
+- `cnames`: `Vector{String}` of the column (or coefficient) names
+- `levels`: `Vector` of the levels of the grouping factor.  Usually `String`s but not necessarily.
+- `fixed`: `AbstractVector{T}` of the global OLS estimates for the coefficients in `cnames` only.
+- `condmodes`: `AbstractMatrix{T}` of size `length(levels) x length(cnames)` of the conditional means/modes for the random effects
+- `grpest`: similar to `condmodes` but allowing for `Missing` values, giving the within group OLS estimates of the coefficients.
+"""
+struct CoefByGroup{T<:AbstractFloat}
+    cnames::Vector{String}
+    levels::Vector
+    fixed::AbstractVector{T}
+    condmodes::AbstractMatrix{T}
+    grpest::AbstractMatrix{Union{Missing,T}}
+end
+
+"""
+    shrinkage(m)
+
+Return a `NamedTuple{fnames(m), NTuple(k, CoefByGroup)}` of the coefficients by group for the grouping factors
+"""
+function shrinkage(m::LinearMixedModel{T}) where {T}
```

Comment:
Looking at the rest of the plumbing, I would keep most of the structures as is, but add  `group` and `coef` kwargs here. `coef` should be a string that exactly matches something from `coefnames`, so you could have defaults like `coef=first(coefnames(m),2)` and `group=first(fnames(m))`

Reply:
As long as you're the one who figures out how to keep track of all the indices, works for me! :smile: